### PR TITLE
Vectorized transforming 3d vectors. Resolves #5 partially.

### LINKED ISSUE #5: Support vectorized transformations/rotations
Something like:
```c++
Eigen::Matrix<double, 3, Eigen::Dynamic> W_landmarks;
Transformation T_B_W;
// ... fill in.

Eigen::Matrix<double, 3, Eigen::Dynamic> B_landmarks(T_B_W.transform(W_landmarks));
```

### PR DESCRIPTION
**Pull request by tcies**

### CODE REVIEW

File: minkindr/include/kindr/minimal/implementation/rotation-quaternion-inl.h [hunk 1]
```diff
@@ -242,6 +242,13 @@ RotationQuaternionTemplate<Scalar>::rotate(
   return q_A_B_*v;
 }
 
+/// \brief rotate vectors, v
+template<typename Scalar>
+typename RotationQuaternionTemplate<Scalar>::Matrix3X
+RotationQuaternionTemplate<Scalar>::rotateVectorized(
+    const typename RotationQuaternionTemplate<Scalar>::Matrix3X& v) const {
+  return q_A_B_.toRotationMatrix()*v;
```
Comment (simonlynen) [inline minkindr/include/kindr/minimal/implementation/rotation-quaternion-inl.h]: spaces

File: minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h [hunk 2]
```diff
@@ -125,6 +125,13 @@ QuatTransformationTemplate<Scalar>::transform(
 }
 
 template<typename Scalar>
+typename QuatTransformationTemplate<Scalar>::Matrix3X
+QuatTransformationTemplate<Scalar>::transform(
+    const typename QuatTransformationTemplate<Scalar>::Matrix3X& rhs) const {
+  return q_A_B_.rotateVectorized(rhs).colwise() + A_t_A_B_;
```
Comment (simonlynen) [inline minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h]: what happens if the matrix has zero columns?

**Comment by tcies** [comment]
Hang on, ambiguous overload for transform(block) as well, need to call transformVectorized and remove * operator.

File: minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h (hunk 2 above)
Comment (tcies) [inline minkindr/include/kindr/minimal/implementation/quat-transformation-inl.h]: Nothing should happen. Will add test.

**Comment by tcies** [comment]
ptal

**Comment by simonlynen** [comment]
crash on empty matrix.

lgtm once tests pass.

**Comment by simonlynen** [comment]
@tcies it is fine to add a CHECK for cols() > 0. Please add a death test for that condition.

**Comment by tcies** [comment]
Can I  disagree? What if some use case can have empty Matrices from time to time? I would find an empty check unnecessarily restrictive. Eigen doesn't do it either, anyways.

**Comment by simonlynen** [comment]
Yes of course you can disagree. :) But I disagree too, since the multiplication of an empty matrix is a bug on the user side and in your method Eigen causes a segfault. I rather have a clean CHECK-fail instead.

**Comment by simonlynen** [comment]
@tcies also if you compile Eigen in debug this will assert cols() > 0 in the multiplication / addition

**Comment by tcies** [comment]
Ok

**Comment by tcies** [comment]
Ha, I didn't even see the segfault. Embarassing.

**Pull request merged**